replica/database.cc

```diff
@@ -1989,7 +1997,7 @@ future<> database::do_apply(schema_ptr s, const frozen_mutation& m, tracing::tra
             co_await coroutine::return_exception(std::runtime_error("view update generator not plugged to push updates"));
         }
 
-        auto lock_f = co_await coroutine::as_future(cf.push_view_replica_updates(_view_update_generator, s, m, timeout, std::move(tr_state), get_reader_concurrency_semaphore()));
+        auto lock_f = co_await coroutine::as_future(cf.push_view_replica_updates(_view_update_generator, s, m, timeout, std::move(tr_state), get_reader_concurrency_semaphore(true)));
```

Comment:
Now I'm confused... Does it mean we were always using the real reader_concurrency_semaphore for the read-before-write, which if I understand is initiated here?

If so, how did we move from the problem of "user reads are slowed down" (the original report in https://github.com/scylladb/scylladb/issues/8873) to the modern "rmw reads aren't slowed down so they cause OOM"?

Reply:
To add to my confusion, I see there is another variant of generating view updates below, in replica/table.cc (the one here is replica/database.cc). I lost track of what's the difference :-(